### PR TITLE
admin: Add Chart Catalog infrastructure

### PR DESCRIPTION
**Pull request by sploiselle**
To support creating a catalog of predefined charts to aid in debugging, this PR introduces the
internal mechanisms by which we can define and generate a catalog of charts.

This PR does not contain an exhaustive catalog, but instead only an illustrative subset of the
catalog to demonstrate the code's ergonomics. I will add the remainder of the catalog once
this code is reviewed.

The provided tests will also be expanded on in a subsequent commit.

@danhhz The library of consts that you proposed grew really unwieldy given the size of the final catalog (~150 distinct values). I'd love to make sure this is as easy as possible to use, though, so please include any other suggestions.

@mrtracy Can I get you to take a peek and ensure that `pkg/ts/catalog/chart_catalog.go`'s [`chartDefaultsPerMetricType`](https://github.com/cockroachdb/cockroach/compare/master...sploiselle:chart-catalog-endpoint?expand=1#diff-fc032d10a16db3caf6e19e81a2549edeR57) has sensible defaults?

Release note: none

### CODE REVIEW

**Comment by sploiselle** [comment]
@spencerkimball Here's the actual "catalog" onto which you should be able to graft this notion of "components." Obviously, let me know if there's a more optimal way to structure this to facilitate that or if there's anyone else I should tag to review it.

@vilterp Do you mind taking a look at this given that it does interface with the admin UI in some capacity?

File: pkg/ts/catalog/catalog_generator.go [hunk 1]
```diff
@@ -0,0 +1,505 @@
...
+	"github.com/pkg/errors"
+
+	prometheusgo "github.com/prometheus/client_model/go"
+)
+
+type sectionDescription struct {
+	// Organization identifies the sections in which to place the chart.
+	// (Sections are created ad hoc as ChartSections whenever they're identified in
+	// a chartDescription.)
+	// Inner array is Level 0 (req), Level 1 (req), Level 2 (opt).
```
Comment (spencerkimball) [inline pkg/ts/catalog/catalog_generator.go]: Not sure what these levels mean. Could you augment the comment?

File: pkg/ts/catalog/chart_catalog.proto [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+  required string axisLabel = 8 [(gogoproto.nullable) = false];
+  // percentiles specifies whether the chart should have its metrics broken
+  // out into percentiles; applies only to histograms.
+  required bool percentiles = 9 [(gogoproto.nullable) = false];
+  // metrics specifies the metics the chart should display.
+  repeated ChartMetric metrics = 10 [(gogoproto.nullable) = false];
+}
+
+// ChartSections organize charts into groups that contain slices or charts and 
+//other subsections.
```
Comment (spencerkimball) [inline pkg/ts/catalog/chart_catalog.proto]: s/\/\/other/\/\/ other/

**Review by spencerkimball — APPROVED** [review]
LGTM. My biggest concern is how we'll encourage people to keep chart_catalog.go up to date.

File: pkg/ts/catalog/catalog_generator.go (hunk 1 above)
Comment (sploiselle) [reply pkg/ts/catalog/catalog_generator.go]: Added much more context to the general structure of the project.

**Comment by sploiselle** [comment]
@spencerkimball If there are any metrics that don't appear in at least one chart, there is a test that fails. Hoping that doesn't disincentivize creating metrics in the first place, but seems reasonable to ensure that the catalog doesn't wither in the face of progress.

**Pull request merged**